torchbenchmark/models/BERT_pytorch/__init__.py

```diff
@@ -33,13 +78,55 @@ def __init__(self, device=None, jit=False):
         ]) # Avoid reading sys.argv here
         args.with_cuda = self.device == 'cuda'
         args.script = self.jit
-        vocab = WordVocab.load_vocab(args.vocab_path)
 
+        # Example effect of batch size on eval time(ms)
+        # bs     cpu       cuda
+        # 1      330       15.5
+        # 2      660       15.5
+        # 4     1200       15.2
+        # 8     2200       20
+        # 16    4350       33
+        # 32    8000       58
+        #
+        # Issue is that with small batch sizes the gpu is starved for work.
+        # Ideally doubling work would double execution time.
+
+        # parameters for work size
+        args.batch_size = 16
+        vocab_size = 20000
+        args.corpus_lines = 50000
+
+        # generate random corpus from parameters
+        self.reset_seeds(1337)
+        vocab = WordVocab(CorpusGenerator(vocab_size, args.corpus_lines))
+
+        #with open(args.train_dataset, "r", encoding="utf-8") as f:
+        #  vocab = WordVocab(f)
+        #vocab = WordVocab.load_vocab(args.vocab_path)
+
+        if False:
+            print("seq_len:")
+            print(args.seq_len)
+            print("batch size:")
+            print(args.batch_size)
+            print("layers")
+            print(args.layers)
+            print("args hidden:")
+            print(args.hidden)
+            print("len vocab:")
+            print(len(vocab))
+            print(type(vocab))
+
+        self.reset_seeds(1337)
```

Comment:
Why reset_seeds multiple times here and on line 100, 124?

Reply:
Its so when data size changes the init of earlier dataset wont change the first part of next dataset. Keeps distribution the same. It was useful when developing the dataset and setting the parameters.